src/blog/01_26_propositions_and_predicates.md

```diff
@@ -0,0 +1,164 @@
+
+# Introduction to Predicates
+
+
+## Recap of Propositions:
+
+For the purpose of propositional logic, **propositions** are (almost) atomic black-boxes:
+They have _names_ and _definitions_, but their names are just atomic symbols in the logic syntax,
+and their definitions are _outside_ the logic syntax (usually these are statements in a human language).
+
+The set of all propositions, together with the special symbols ⊤ and ⊥,
+are the atomic symbols of the syntax of **formulas**.
+The connectives will be familiar from their roles as boolean operators: ¬, ∧ ,∨, →, _etc_.
+Thus, formulas are recursive trees with connectives as nodes and atomic symbols as leaves. 
+
+There are two truth values: 1 and 0.
+(These are not part of the syntax of formulas.)
+These can go by other names, commonly _"true"_ and _"false"_.
+The properties and isomorphisms of [Boolean Algebra](https://en.wikipedia.org/wiki/Boolean_algebra)
+are all good fun, but for now we can simply take **truth values** ≜ {1, 0}.
+
+The operator **⟦·⟧** maps formula to truth values; _e.g._ ⟦⊤⟧=1.
+(This example is actually part of the definition of ⟦·⟧;
+⊤ and ⊥ are given meaning respectively as an atomic tautology and an atomic contradiction.)
+Although it's often left implicit, it technically has a subscript argument,
+either an **assignment** (a set of (_proposition_, _truth value_) pairs)
+or a pair (_KB_, σ) of a **knowlege base** (a list of statements of the same form as proposition definitions)
+and a **storage** of proposition definitions.
+In either case, ⟦·⟧₋ can be implemented in terms of lookups and if-then-else statements. 
+
+
+## Predicates as Relations:
+
+Consider the category _B_ of truth values, and whatever category _S_ of nouns/subjects.
+A single proposition (_e.g._ _p_≜`"Six is prime."`) can be represented as a pair of form 
+(_s_∈_S_, _R_) where _R_ is a [relation](https://en.wikipedia.org/wiki/Binary_relation) over _S_ and _B_
+(_e.g._ (6, _IsPrime_)).
+
+Not any relation _R_⊆_S_⨯_B_ will serve in the representation of a proposition!
+It must be a **predicate**\*, a relation mapping _every_ element of _S_ to exactly one truth value.
```

Comment:
This is at the edge of my knowledge, but it looks like you are suggesting that predicates must be total. If yes, I am not sure I've ever seen this stated explicitly. The reason I bring it up is  because a suitable structure for FOL is defined to (amongst other things) make the predicate total. I don't know if there is every any utility for unsuitable structures. You could imagine someone proffering a structure for a FOL formula that does not define predicates used in tautologies or contradictions (assuming they aren't used elsewhere), but then again they could always minimize the formula.  It's very much an edge case and there is no reason to not assume predicates are bijections onto truth values (and if you've seen it stated explicitly this way, cool!); I just didn't introduce it in class as such.

@cibelemf : pinging your expertise, if you are interested.

Reply:
I do not understand the plurality of logics well enough to consider the implications of total vs partial predicates. Predicates seem to be frequently defined as a kind of function, and functions are total unless otherwise specified (right?), so I just assumed.

Reply:
For predicates, we are almost always going to start by saying that a structure is suitable and a suitable structure, by definition, is going to make the mapping of all predicates and functions total. 

In the general case though functions are only required to be partial (classically: square root over reals, right?). I'm sure there many sub-disciplines where functions are nearly always total (and thinking about it now, this is nearly always the case in PL/type theory land -- hence option types!).  I wouldn't assume functions are total unless it's clear from context.

Reply:
I added this to the footnote.